x/stake/endblock.go

```diff
@@ -31,9 +40,26 @@ func EndBlocker(ctx sdk.Context, k keeper.Keeper) (validatorUpdates []abci.Valid
 			panic("sideChainIds does not equal to sideChainStores")
 		}
 	}
+
+	if sdk.IsUpgrade(sdk.BCFusionSecondHardFork) {
+		for i := range storePrefixes {
+			sideChainCtx := ctx.WithSideChainKeyPrefix(storePrefixes[i])
+			refundEvents.AppendEvents(handleRefundStake(sideChainCtx, k))
+			_, _, events = endRefundBlock(ctx, k)
+			refundEvents.AppendEvents(events)
+		}
+
+		refundEvents.AppendEvents(handleRefundStake(ctx, k))
```

Comment:
Please remove these code, as BC delegation no need to be undelegated.

Reply:
fixed